### ISSUE TITLE
Examples have incorrect subpackages

### ISSUE THREAD

**Issue by dkorpel** [rtbo/gfx-d#4]
I want to try the examples, but building is non-trivial and I haven't found instructions.
So first of all, a recursive clone is needed so that `3rdparty/img-d` isn't empty. 
Next up, dub says `Sub package "gfx:window" doesn't exist.`. The cause seems to be this line in dependencies:
```
"gfx:window" : { "path": "../../window" },
```
The subpackage is defined in the root, not in the window folder. I tried to solve this, also for the gl3 and vulkan dependencies, but I either got this when changing the name:
`Sub package vulkan must be referenced using the path to it's parent package.`
Of this when changing paths:
```
Unresolvable dependencies to package gfx:
  texture ~master depends on gfx >=0.0.0 @(...)\gfx-d\
  texture ~master depends on gfx:genmesh >=0.0.0 @(...)\gfx-d\genmesh
```
I don't know what the package structure is supposed to be but I the current one isn't working for me.

**Comment by rtbo** [comment]
Ran the following, and had errors I never had before! 
Dub config looks fine however.

```sh
$ rm -rf ~/.dub  # clearing dub cache and local config
$ git clone --recursive git@github.com:rbo/gfx-d.git
$ cd gfx-d
$ dub run gfx:triangle
Building package gfx:triangle in /home/remi/dev/gfx-d/examples/triangle/
Warning: This package uses the "noDefaultFlags" build requirement. Please use only for development purposes and not for released packages.

Warning: This package uses the "noDefaultFlags" build requirement. Please use only for development purposes and not for released packages.

Fetching derelict-util 2.0.6 (getting selected version)...
Fetching spirv_cross 0.3.0 (getting selected version)...
Fetching wayland 0.1.0 (getting selected version)...
Fetching xcb-d 2.1.1+1.11.1 (getting selected version)...
Fetching xlib-d 0.1.1 (getting selected version)...
Warning: This package uses the "noDefaultFlags" build requirement. Please use only for development purposes and not for released packages.

Running pre-generate commands for wayland:client...
Building package wayland:scanner in /home/remi/.dub/packages/wayland-0.1.0/wayland/scanner/
Performing "release" build using /usr/bin/dmd for x86_64.
wayland:scanner 0.1.0: building configuration "application"...
Linking...
Running ./scanner/wayland_scanner -c client -m wayland.client.protocol -i /home/remi/.dub/packages/wayland-0.1.0/wayland/client/../protocol/wayland.xml -o /home/remi/.dub/packages/wayland-0.1.0/wayland/client/source/wayland/client/protocol.d
Running pre-generate commands for gfx:window...
Generating gfx.window.wayland.zxdg_shell_v6 from /home/remi/dev/gfx-d/window/resources/xdg-shell-unstable-v6.xml
Building package wayland:scanner in /home/remi/.dub/packages/wayland-0.1.0/wayland/scanner/
Performing "$DFLAGS" build using /usr/bin/dmd for x86_64.
wayland:scanner 0.1.0: building configuration "application"...
Linking...
Running ../../.dub/packages/wayland-0.1.0/wayland/scanner/wayland_scanner -m gfx.window.wayland.zxdg_shell_v6 -i /home/remi/dev/gfx-d/window/resources/xdg-shell-unstable-v6.xml -o /home/remi/dev/gfx-d/window/source/gfx/window/wayland/zxdg_shell_v6.d
Running pre-generate commands for gfx:triangle...
/home/remi/dev/gfx-d/examples/triangle/views/shader.vert.spv is up-to-date
/home/remi/dev/gfx-d/examples/triangle/views/shader.frag.spv is up-to-date
Performing "debug" build using /usr/bin/dmd for x86_64.
gfx:core 0.1.1: building configuration "library"...
gfx:math 0.1.1: building configuration "library"...
gfx 0.1.1: building configuration "library"...
gfx:bindings 0.1.1: building configuration "library"...
spirv_cross 0.3.0: building configuration "library"...
derelict-util 2.0.6: building configuration "library"...
xcb-d 2.1.1+1.11.1: building configuration "library"...
xlib-d 0.1.1: building configuration "library"...
gfx:gl3 0.1.1: building configuration "linux"...
wayland:client 0.1.0: building configuration "static"...
gfx:vulkan 0.1.1: building configuration "linux"...
gfx:window 0.1.1: building configuration "linux"...
gfx:triangle 0.1.1: building configuration "application"...
Linking...
/usr/bin/ld: ../../.dub/packages/wayland-0.1.0/wayland/client/.dub/build/static-debug-linux.posix-x86_64-dmd_2083-16307D3C86D6D519814CEFD66FF69CFE/libwayland_client.a(package.o):(.data.rel.ro+0x18): undefined reference to `_D7wayland6client8protocol12__ModuleInfoZ'
/usr/bin/ld: ../../.dub/packages/wayland-0.1.0/wayland/client/.dub/build/static-debug-linux.posix-x86_64-dmd_2083-16307D3C86D6D519814CEFD66FF69CFE/libwayland_client.a(core.o):(.data.rel.ro+0x10): undefined reference to `_D7wayland6client8protocol12__ModuleInfoZ'
/usr/bin/ld: ../../.dub/packages/wayland-0.1.0/wayland/client/.dub/build/static-debug-linux.posix-x86_64-dmd_2083-16307D3C86D6D519814CEFD66FF69CFE/libwayland_client.a(core_303_514.o): in function `_D7wayland6client4core13WlDisplayBase7connectFxAyaZCQByQBt8protocol9WlDisplay':
/home/remi/dev/gfx-d/../../.dub/packages/wayland-0.1.0/wayland/client/source/wayland/client/core.d:89: undefined reference to `_D7wayland6client8protocol9WlDisplay7__ClassZ'
/usr/bin/ld: /home/remi/dev/gfx-d/../../.dub/packages/wayland-0.1.0/wayland/client/source/wayland/client/core.d:89: undefined reference to `_D7wayland6client8protocol9WlDisplay6__ctorMFPSQBt6nativeQBv10wl_displayZCQCuQCpQClQCf'
/usr/bin/ld: ../../.dub/packages/wayland-0.1.0/wayland/client/.dub/build/static-debug-linux.posix-x86_64-dmd_2083-16307D3C86D6D519814CEFD66FF69CFE/libwayland_client.a(core_303_514.o): in function `_D7wayland6client4core13WlDisplayBase11connectToFdFxiZCQCbQBw8protocol9WlDisplay':
/home/remi/dev/gfx-d/../../.dub/packages/wayland-0.1.0/wayland/client/source/wayland/client/core.d:95: undefined reference to `_D7wayland6client8protocol9WlDisplay7__ClassZ'
/usr/bin/ld: /home/remi/dev/gfx-d/../../.dub/packages/wayland-0.1.0/wayland/client/source/wayland/client/core.d:95: undefined reference to `_D7wayland6client8protocol9WlDisplay6__ctorMFPSQBt6nativeQBv10wl_displayZCQCuQCpQClQCf'
collect2: error: ld a retourné le statut de sortie 1
Error: linker exited with status 1
/usr/bin/dmd failed with exit code 1.
```

**Comment by dkorpel** [comment]
Strange, `dub run gfx:triangle` doesn't work for me. I'm on Windows 10.
```
$ git pull
Already up to date.

$ ls
3rdparty/  core/  dub.json   genmesh/  gl3/   memalloc/  source/  vulkan/
bindings/  decl/  examples/  gfx.lib   math/  README.md  tools/   window/

$ dub --version
DUB version 1.12.0, built on Nov 21 2018

$ dub run gfx:triangle
Failed to find a package named 'gfx:triangle'.
```

**Comment by dkorpel** [comment]
Nevermind, I edited the dub files so I needed to do a `git checkout .`. I do have a problem with building SPIRV though:
```
-- The C compiler identification is MSVC 19.0.23026.0
-- The CXX compiler identification is MSVC 19.0.23026.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.11/Modules/CMakeTestCCompiler.cmake:52 (message):
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/cl.exe"

  is not able to compile a simple test program.
```

**Comment by rtbo** [comment]
I've pushed spirv_cross-d-v0.3.1. It includes pre-built windows binaries, so this error should go away.

**Comment by rtbo** [comment]
OK I think I fixed this with `gfx-d-0.1.2`. Please checkout latest master and tell me.

The following also works on my windows machine:
```sh
$ dub fetch gfx
$ dub run gfx:triangle
```
(this won't work for examples that require `img-d`)

Note however that examples do not run so well on Windows than Linux.

**Comment by dkorpel** [comment]
Sorry for neglecting this issue. Revisiting it today, dub successfully compiles the example projects when following the readme, so thanks!
